test/functional/rpc_blockchain.py

```diff
@@ -255,6 +256,13 @@ def _test_getdeploymentinfo(self):
         # calling with an explicit hash works
         self.check_signalling_deploymentinfo_result(self.nodes[0].getdeploymentinfo(gbci207["bestblockhash"]), gbci207["blocks"], gbci207["bestblockhash"], "started")
 
+    def _test_y2106(self):
+        self.generate(self.nodes[0], 8)[-1]
+        time_2106 = 2**32 - 1
+        self.nodes[0].setmocktime(time_2106)
+        last = self.generate(self.nodes[0], 6)[-1]
+        assert_equal(self.nodes[0].getblockheader(last)["mediantime"], time_2106)
```

Comment:
Might as well generate one more block to trigger the (other) 2106 bug: `time-too-old, block's timestamp is too early`

Reply:
I'd rather not, because this would require the whole file to be ignored by the `implicit-signed-integer-truncation` sanitizer:

```
 node0 stderr node/miner.cpp:36:25: runtime error: implicit conversion from type 'int64_t' (aka 'long') of value 4294967296 (64-bit, signed) to type 'uint32_t' (aka 'unsigned int') changed the value to 0 (32-bit, unsigned)